lib/travis/api/app/endpoint/jobs.rb

```diff
@@ -140,6 +143,17 @@ def archive_url(path)
       def hostname(name)
         "#{name}#{'-staging' if Travis.env == 'staging'}.#{Travis.config.host.split('.')[-2, 2].join('.')}"
       end
+
+      private def include_log_id?
+        params[:include_log_id] ||
+          !Travis.config.logs_api.enabled? ||
+          request.user_agent.to_s.start_with?('Travis')
+      end
+
+      private def empty_log(job_id)
+        # XXX: Should this empty version be defined elsewhere?
```

Comment:
perhaps the service should have this logic - so it returns an empty log if it can't find one. So move the empty log logic down into the service or RemoteLog class.